testing/endtoend/components/validator.go

```diff
@@ -378,3 +396,49 @@ func sendDeposits(web3 *ethclient.Client, keystoreBytes []byte, num, offset int,
 	}
 	return nil
 }
+
+func createProposerSettingsPath(pubkeys []string, validatorIndex int) (string, error) {
+	testNetDir := e2e.TestParams.TestPath + fmt.Sprintf("/proposer-settings/validator_%d", validatorIndex)
+	configPath := filepath.Join(testNetDir, "config.json")
+	if len(pubkeys) == 0 {
+		return "", errors.New("number of validators must be greater than 0")
+	}
+	var proposerSettingsPayload validator_service_config.ProposerSettingsPayload
+	if len(pubkeys) == 1 {
+		proposerSettingsPayload = validator_service_config.ProposerSettingsPayload{
+			DefaultConfig: &validator_service_config.ProposerOptionPayload{
+				FeeRecipient: DefaultFeeRecipientAddress,
+			},
+		}
+	} else {
+		config := make(map[string]*validator_service_config.ProposerOptionPayload)
+
+		for _, pubkey := range pubkeys {
+			// Create an account
+			key, err := crypto.GenerateKey()
```

Comment:
there is a simplification you can use here. You dont need to generate a new key, what you can do is just take the first 20-bytes of the bls key and use that as the fee recipient. You won't have possession of the key, but the fee recipient evaluator just needs to check that the balance increment has gone to the correct address.